aws/cerberus/cerberus.tf

```diff
@@ -0,0 +1,582 @@
+
+  # Enables termination protection. Must be set to false and applied before inctance can be removed.
+  disable_api_termination = "${var.termination_protection}"
+  monitoring              = true
+  ebs_optimized           = true
+
+  tags = "${merge(local.tags, map("Name", format("${local.module_prefix}-%d", count.index + 1), "Schedule", format("%s", var.schedule)))}"
+
+  lifecycle {
+    ignore_changes = ["tags.%", "tags.Schedule", "tags.ScheduleStatus", "tags.ScheduleTimestamp"]
+  }
+}
+
+resource "aws_ssm_association" "domain_join" {
+  count = "${var.create && var.add_instance_to_ad == "true" ? var.number_of_instances : 0}"
+  name  = "AWS-JoinDirectoryServiceDomain"
+
+  instance_id = "${aws_instance.cerberus_ec2.*.id[count.index]}"
+
+  parameters {
+    directoryId   = "${data.terraform_remote_state.vpc.ds_directory_id}"
+    directoryName = "${data.terraform_remote_state.vpc.ds_domain_name}"
+    directoryOU   = "${var.directory_ou}"
+  }
+}
+
+resource "aws_route53_record" "cerberus_ec2" {
+  count    = "${var.create == "true" ? var.number_of_instances : 0}"
+  provider = "aws.master"
+
+  zone_id = "${data.terraform_remote_state.vpc.vpc_dns_zone_id}"
+  name    = "${var.name}-${format("%d", count.index + 1 )}"
+  type    = "CNAME"
+  ttl     = "30"
+  records = ["${aws_instance.cerberus_ec2.*.private_dns[count.index]}"]
+}
+
+# ALB & resources
+
+resource "aws_security_group" "cerberus_alb" {
+  count       = "${var.create == "true" ? 1 : 0 }"
+  name        = "${local.module_prefix}-alb"
+  description = "${var.desc_prefix} Controls traffic to ALB"
+  vpc_id      = "${data.terraform_remote_state.vpc.vpc_id}"
+
+  ingress {
+    from_port   = "80"
+    to_port     = "80"
+    protocol    = "6"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "${var.desc_prefix} HTTP"
+  }
+
+  ingress {
+    from_port   = "443"
+    to_port     = "443"
+    protocol    = "6"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "${var.desc_prefix} HTTPS"
+  }
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+    description = "${var.desc_prefix} ALB egress"
+  }
+
+  tags = "${merge(local.tags, map("Name", "${local.module_prefix}-alb"))}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_lb" "cerberus_alb" {
+  count = "${var.create && var.enable_https == "true" ? 1 : 0 }"
+
+  name               = "${local.module_prefix}-alb"
+  internal           = "false"
+  load_balancer_type = "application"
+  security_groups    = ["${aws_security_group.cerberus_alb.id}"]
+  subnets            = ["${data.terraform_remote_state.vpc.vpc_public_subnets}"]
+  idle_timeout       = "60"
+
+  tags = "${local.tags}"
+}
+
+resource "aws_lb_target_group" "cerberus_alb_target_group" {
+  count = "${var.create && var.enable_https == "true" ? 1 : 0 }"
```

Comment:
Does this work? I would have assumed this would look like this `${var.create == "true" && var.enable_https == "true" ? 1 : 0 }`